### PR TITLE
DID rewind

### LINKED ISSUE #692: New Orgs / Nodes Rejecting Previous Node Identity Claims
## Description

When creating multi-org networks via the Helm chart, we would stand up the first node for an org, then stand up a new node / org, then another and so on (a bit different from how the CLI creates networks). In the new nodes, we would observe them rejecting the first org's node identity claim messages due to:

```
[2022-04-11T17:19:21.074Z]  WARN poc4-par2-firefly: Unable to process identity claim 564be456-7834-4218-84b9-1a82c9011270: FF10214: Identity '8e8b6544-df58-43fe-935e-f0078b760e7d' not found in identity chain for did:firefly:node/poc4-par1-firefly 'd81e6067-bb73-4613-83c3-49b4a0a08051' dbtx=85C5Fddx role=aggregator
```

The ethconnects were in catchup mode, so the other nodes were presumed to be processing the org and node identity claim messages in the same batchpin. So it appears to be a bug in the logic where the aggregator does not account for needing to process org identity claims before node identity claims within the same batch. 

## Other Info

FireFly Version: `1.0.0-rc3`
Ethconnect Version: `v3.1.5`
Blockchain Provider: `Quorum v21.10.0 (Geth v1.9.25)`

Relevant Ethconnect Configuration:

```yaml
rest:
  rest-gateway:
    leveldb:
      path: "/qdata/ethconnect/receiptsdb"
      queryLimit: 100
    alwaysManageNonce: false
    maxTXWaitTime: 120
    openapi:
      storagePath: "/qdata/ethconnect/openapi"
      eventsDB: "/qdata/ethconnect/eventsdb"
      catchupModeBlockGap: 10000
      catchupModePageSize: 10000
```

## Logs

First Org / Node : [poc4-par1.log](https://github.com/hyperledger/firefly/files/8466801/poc4-par1.log)

Other Orgs / Nodes:
* [poc4-par2.log](https://github.com/hyperledger/firefly/files/8466802/poc4-par2.log)
* [poc4-par3.log](https://github.com/hyperledger/firefly/files/8466803/poc4-par3.log)

### PR DESCRIPTION
**Pull request by peterbroadhurst**
Fixes #692 
In PR chain with #707

This PR adds a new reason for a rewind - to go back and confirm messages that were previously parked because the author was not known.

We cannot assure that the identity verification for an identity, will be processed by the aggregator before messages sent by that identity, or child identities registered under that identity. Because they are all sent on different topics.

### CODE REVIEW

File: internal/definitions/definition_handler_identity_claim_test.go [hunk 1]
```diff
@@ -177,13 +177,14 @@ func TestHandleDefinitionIdentityClaimCustomWithExistingParentVerificationOk(t *
 	action, err := dh.HandleDefinitionBroadcast(ctx, bs, claimMsg, fftypes.DataArray{claimData}, fftypes.NewUUID())
 	assert.Equal(t, HandlerResult{Action: ActionConfirm}, action)
 	assert.NoError(t, err)
+	assert.Equal(t, bs.confirmedDIDClaims, []string{custom1.DID})
```
Comment (nguyer) [inline internal/definitions/definition_handler_identity_claim_test.go:180]: I'm not sure I really trust these `bs` claims...

File: internal/events/aggregator_rewind.go [hunk 2]
```diff
@@ -283,3 +293,27 @@ func (rw *rewinder) getRewindsForBlobs(ctx context.Context, newHashes []driver.V
...
+
+func (rw *rewinder) getRewindsForDIDs(ctx context.Context, dids []driver.Value, batchIDs map[fftypes.UUID]bool) error {
+
+	// We need to find all pending messages, that are authored by this DID
+	fb := database.MessageQueryFactory.NewFilter(ctx)
+	filter := fb.And(
+		fb.Eq("state", fftypes.MessageStatePending),
+		fb.In("author", dids),
+	)
+	records, err := rw.database.GetMessageIDs(ctx, filter)
```
Comment (nguyer) [inline internal/events/aggregator_rewind.go:305]: Is there a risk here that this becomes a large query and needs to be paginated?

File: manifest.json [hunk 3]
```diff
@@ -44,6 +44,6 @@
     "release": "v0.7.0_50810a4"
   },
   "cli": {
-    "tag": "v0.0.46"
+    "tag": "6fb3585a8e279ff9860f3910625cb5b0ee56c749"
```
Comment (nguyer) [inline manifest.json:47]: Should be changed to an actual release when `v0.0.48` is available

**Review by nguyer — APPROVED** [review]

File: internal/events/aggregator_rewind.go (hunk 2 above)
Comment (peterbroadhurst) [reply internal/events/aggregator_rewind.go:305]: I did consider that, and determined the answer should be no (for now):
- We are only querying the id+sequence, so the in-memory size is very small per record
- We are only querying things from a given author that has just become confirmed, so we should be dealing with a small number of records

**Pull request merged**